sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/DefaultSource.scala

```diff
@@ -61,7 +61,9 @@ class DefaultSource extends FileFormat with DataSourceRegister {
     val firstRow = new LineCsvReader(csvOptions).parseLine(firstLine)
 
     val header = if (csvOptions.headerFlag) {
-      firstRow
+      firstRow.zipWithIndex.map { case (value, index) =>
+        if (value == "" || value == null) s"C$index" else value
```

Comment:
I mean if the header is a empty string then, I think the field name should be a empty string since apparently it works with fields named empty strings. I tested this by manually giving a schema. If the header is used for schema, then I think the names should be as they are. We don't really change field names specified in ORC, Parquet or JSON.